# HG changeset patch
# User Rik <rik@octave.org>
# Date 1465422492 25200
#      Wed Jun 08 14:48:12 2016 -0700
# Node ID cf690a73a186253d4ada212fec898c9033504384
# Parent  f382293bc11da3572c323429a52c27aea9ed94b3
uiputfile.m: Correctly handle '*.*' filter type (bug #48182).

* uiputfile.m: Only append extension if it is not present in filename,
but was specified in filter.  Detect special case of '*.*' and avoid
adding '.' as extension.

diff --git a/scripts/gui/uiputfile.m b/scripts/gui/uiputfile.m
--- a/scripts/gui/uiputfile.m
+++ b/scripts/gui/uiputfile.m
@@ -111,21 +111,22 @@ function [retfile, retpath, retindex] = 
     [retfile, retpath, retindex] = __octave_link_file_dialog__ (outargs{:});
   else
     funcname = __get_funcname__ (mfilename ());
     [retfile, retpath, retindex] = feval (funcname, outargs{:});
   endif
 
   ## Append extension to the name if it isn't already added.
   if (ischar (retfile))
-    ext = outargs{1}{retindex};
-    ext = strrep (ext, "*", "");
-
-    if (length (retfile) >= length (ext))
-      if (! strcmp (retfile(end-length (ext)+1:end), ext))
+    [~, ~, ext] = fileparts (retfile);
+    if (isempty (ext))
+      ext = outargs{1}{retindex};
+      ext = strrep (ext, "*", "");
+      if (ext != '.')
+        [~, ~, ext] = fileparts (ext);  # paranoid checking of extension
         retfile = [retfile ext];
       endif
     endif
   endif
 
 endfunction
 
 
